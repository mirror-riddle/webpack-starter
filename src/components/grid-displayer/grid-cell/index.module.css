.cell {
  display: flex;
  justify-content: center;
  align-items: center;
  border: dashed 1px rgba(0, 0, 0, 0.1);
}
