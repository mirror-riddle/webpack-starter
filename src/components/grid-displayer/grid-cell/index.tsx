import classNames from 'classnames';
import React from 'react';

import styles from './index.module.css';

interface GridCellProps {
  className?: string;
}

const GridCell: React.FunctionComponent<GridCellProps> = props => {
  const { className, children } = props;
  const cellClassName = classNames(className, styles.cell);

  return (
    <div className={cellClassName}>
      <div className={styles.content}>{children}</div>
    </div>
  );
};

export { GridCell };
