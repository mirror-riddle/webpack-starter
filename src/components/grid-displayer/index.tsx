import React from 'react';
import classNames from 'classnames';

import { GridCell } from './grid-cell';

import styles from './index.module.css';

interface GridDisplayerProps {
  className?: string;
  cellCount?: number;
  style?: Record<string, string>;
}

const GridDisplayer: React.FunctionComponent<GridDisplayerProps> = props => {
  const { className, cellCount, style } = props;
  const gridClassName = classNames(className, styles.grid);
  const cellArray = Array.from(Array(cellCount || 20).keys());
  return (
    <div className={gridClassName} style={style}>
      {cellArray.map(cell => (
        <GridCell key={cell}>{cell}</GridCell>
      ))}
    </div>
  );
};

export { GridDisplayer };
