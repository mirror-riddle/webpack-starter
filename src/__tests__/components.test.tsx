import React from 'react';
import Renderer from 'react-test-renderer';

import { GridCell } from 'components/grid-displayer/grid-cell';

describe('components tests: ', () => {
  test('GridItem className contains cell', () => {
    const json = Renderer.create(<GridCell className="cell" />).toJSON();
    expect(json).toMatchSnapshot();
  });
});
