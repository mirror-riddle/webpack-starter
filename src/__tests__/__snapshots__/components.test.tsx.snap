// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test components render <GridItem /> 1`] = `
<div
  className="cell item"
>
  <div
    className="content"
  />
</div>
`;

exports[`test components render <RandomKey /> 1`] = `
<div
  className="content"
>
  <pre
    className="pre"
  >
    <code>
      
    </code>
  </pre>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      change
    </span>
  </button>
</div>
`;
